### PR TITLE
Fix delay time false negative

### PR DESCRIPTION
**Pull request by bspaulding**
Added a new test file, animated with a 0 delay time. This was producing a false negative.

There also appears to be a failing test on master, and the `generic-header.gif` file exists in the npm package, but looks like it got deleted at some point.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,11 +25,14 @@ AnimatedGifDetector.prototype.isAnimated = function(buffer) {
   var result = false
     , count = 0
   ;
-  for (var i = 0; i < buffer.length; i++) {
+  var packed = buffer.slice(10, 11)[0];
+  var colorTableSize = 3*2^(packed+1);
+  var leadBytes = 13 + colorTableSize;
+  for (var i = leadBytes + 1; i < buffer.length; i++) {
```
Comment (bspaulding) [inline index.js]: Can remove this if you like, but it just calculates how big the global color table is, and doesn't bother searching through that header space.

**Comment by tbuchok** [comment]
@bspaulding thanks! mind if i take a look in the morning? all sounds great from your description - appreciate it!

**Comment by bspaulding** [comment]
of course, no rush

**Comment by tbuchok** [comment]
published to npm [animated-gift-detector@1.1.3](https://www.npmjs.com/package/animated-gif-detector)

thanks again, bradley.

**Comment by bspaulding** [comment]
Ah, yea. It looks like there is a bug here re: calculating the color table size. It should be using the last 3 bits of the packed byte, rather than the whole byte value. Curious that it still passed all the tests... heh.

File: index.js [hunk 2]
```diff
@@ -30,11 +25,14 @@ AnimatedGifDetector.prototype.isAnimated = function(buffer) {
   var result = false
     , count = 0
   ;
-  for (var i = 0; i < buffer.length; i++) {
+  var packed = buffer.slice(10, 11)[0];
```
Comment (bspaulding) [inline index.js]: re: your comment, this should be that whole packed byte, i believe. which means that this does indeed have an issue.

I can follow on with a fix, but I may have to manufacture a gif to expose the bug, heh.

**Comment by tbuchok** [comment]
you have time to investigate? or:

- should i spend time on this?
- we revert the "skip the color table" stuff?

thoughts?

**Comment by bspaulding** [comment]
I think it makes sense to revert the color table stuff until we have a fix.

**Pull request merged**